src/QuadVelocityController.cpp

```diff
@@ -213,59 +173,63 @@ bool QuadVelocityController::getTransformAfterTime(
         ROS_ERROR("Could not transform map to level_quad: %s",ex.what());
     }
 
-    // ros::ok() must ahve equaled false, program is exiting.
-    ROS_ERROR("ros::ok false while waiting for transform");
+    ROS_ERROR("Exception or ros::ok was false");
     return false;
 }
 
 // Gets the velocity from the transformation information
 // Blocks while waiting for a new transform to calculate the velocity with
-bool QuadVelocityController::getVelocities(geometry_msgs::Twist& return_velocities)
+bool QuadVelocityController::waitForNewVelocities(geometry_msgs::Twist& return_velocities)
 {
 
-    // Get current ROS time
-    ros::Time time = ros::Time::now();
-
-    // Check if we already have a velocity at this time
-    if (have_last_velocity_stamped_ && time == last_velocity_stamped_.header.stamp) {
-        return_velocities = last_velocity_stamped_.twist;
-        return true;
-    }
-
-    // We don't have a velocity at this time yet, so make sure we don't fetch
-    // a transform at the same time as the last one
-    while (ros::ok()
-            && ran_once_
-            && time == last_transform_stamped_.header.stamp) {
-        ros::spinOnce();
-    }
-
-    // Calculate the lastest time between transforms that is allowed.
-    ros::Time latest_time_allowed;
-    if (ran_once_)
+    // Make sure that the current time is more than the last transform stamped time
+    while (wait_for_velocities_ran_once_
```

Comment:
I don't really think we should be doing this because it prevents us from processing future dated transforms if they're available.